card/db/migrate/20150220141031_create_deck_routes_file.rb

```diff
@@ -0,0 +1,14 @@
+# -*- encoding : utf-8 -*-
+
+class CreateDeckRoutesFile < ActiveRecord::Migration
+  def up
+    routes_path = File.join Wagn.root, 'config', 'routes.rb'
+    if !File.exist? routes_path
+      template_path = File.join Wagn.gem_root, 'lib', 'wagn', 'generators', 'wagn', 'templates', 'config', 'routes.erb'
+      @include_jasmine_engine = false
+      decko_path = ''
+      routes = ERB.new(File.read template_path).result binding
+      File.open(routes_path, "w") { |f|  f.write routes }
+    end
+  end
+end
```

Comment:
No, not reversed.

When you create an app deck, it creates an application.rb file (in config) that inherits from Wagn::Application (soon Decko::Application).

When you create an engine deck, it should create an engine.rb file that inherits from Decko::Engine.  When I said "engines don't have an application.rb", I meant in the engine itself.   *Engine decks should generally not require editing the config/application.rb file of the host rails app.*  Imo they should require an edit to the host app's config/routes.rb file and, generally speaking, nothing else.

The engine deck app's engine.rb file should be a lot like an app deck's application.rb file.  It should have a lot of common options demonstrated and commented out.   These templates belong in decko-rails.